### ISSUE TITLE
Refreshing forever on Mac Sierra

### ISSUE THREAD

**Issue by ghost** [zz85/space-radar#38]
Looks nice on the videos, but as far as I can tell this does not work on Mac Sierra (10.12.6). If I scan the disk or a folder or add a du file it always just displays "Refreshing..." (tried two separate Macs). During the scan I can see the "Scanning..." window and it seems to get stuck on a file. I tried earlier versions too (5, 4.4, etc).

**Comment by krisfield** [comment]
I'm seeing the same issue on Mac Sierra (10.12.6)

**Comment by zz85** [comment]
Can I just check if this is happening with v5.1.0? Let me try to replicate this. Thanks for reporting!

**Comment by zz85** [comment]
Hmm unfortunately I'm already on High Sierra, may I know how many GBs of space has been approximately scanned before it seems to have frozen?

**Comment by krisfield** [comment]
Yes it is v5.1.0, when selecting "Scan drive" it claims to have scanned 26.64 GBs when it freezes up but it displays 26.64 GB almost as soon as the scan begins. When selecting "Scan Folder" it only says "Refreshing" even when selecting a small folder, 8.1 MB, it will just hang on the refreshing screen.

**Comment by zz85** [comment]
Hmm this seems strange and sorry for forgetting to get back to you @krisfield... Since I'm not sure how to debug this, perhaps try running du and piping the output to the file first? https://github.com/zz85/space-radar#reading-file 

If spaceradar still fails to load it, I could try loading the file from my end to see if I can identify what the issue is.

**Comment by hydrosquall** [comment]
@krisfield / @parsonsk - I had this issue when using the pre-built binary on Mac, but the infinite loader issue went away when I manually cloned the git repository and ran `npm run app`.

**Comment by Hugo4IT** [comment]
This still happens on my M1 Mac running the latest MacOS Sonoma, it gets stuck on some random file and never continues. The file seems to be completely random, so does the point at which it freezes as I've had this happen at 70Gb, 30Gb, 40Gb. There's only one thing certain and that is that it has never completed once. Any known workaround, or a way I could help fix or at least diagnose this issue?

<img width="879" alt="image" src="https://github.com/zz85/space-radar/assets/48245742/2c0c137d-1350-4b80-9713-dda529a31441">

**Comment by zz85** [comment]
@Hugo4IT unfortunately my guess is that scanning large systems causes memory limit exceeding on the V8 engine on electron. this project started more than 9 years ago has a hack and unfortunately I haven't found the motivation to fix some of the fundamental problems. perhaps one day I might work on a space-radar v2 in rust that would fix some of these issues.
